### PR TITLE
USWDS - Footer: Added styling for WHC Mode

### LINKED ISSUE #4517: Phase 1: Windows HC Fixes
Complete fixes outlined by audit to components in Windows HC mode. 

- [x]  **Accordion** - Outline & expand/collapse icons missing for menu items (which makes it hard for users to tell it is an expandable accordion) - Fixed in #4521
- [x] **Banner** - Chevron missing (without those, users cannot tell they can expand the banner for more information) - Resolved in https://github.com/uswds/uswds/pull/4529
- [x] **Checkbox** - Checkboxes do not show in high contrast mode - Fixed in #4509 
- [x]  **Header** - Fixed in #4485 
- [x] **Radio Buttons** - Radio buttons are invisible and do not show which is selected. - Fixed in #4509 
- [x] **Range Slider** - Slider handle missing an outline and becomes negative space in between the range bar - Fixed in #4510
- [x] **Pagination** - Selected page should have a bold border or different color background. In high contrast it isn't possible to see what page the user is currently on. - Fixed in #4523 
- [x] **Footer** - Social media icons missing. Chevron icon missing from mobile view -  Resolved in #4528

### PR DESCRIPTION
**Pull request by mahoneycm**
## Preview

[Footer →](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/cm-hc-footer/components/detail/footer--default.html)

## Description

Resolves #4517  footer issues 

Currently, social media icons and expand chevrons (mobile only) disappear in dark high contrast modes, causing the user to lose context and content.

This PR also fixes the disappearing chevrons in the big footer variant in mobile views.

### Social Icons

Currently: 

_Alternative color shows hover_

image
_Default_

image
_High Contrast_

After Fix:

image

### Mobile Chevrons

Currently:

image

After Fix:

image

Before you hit Submit, make sure you’ve done whichever of these applies to you:

- [x] Follow the [18F Front End Coding Style Guide](https://pages.18f.gov/frontend/) and [Accessibility Guide](https://pages.18f.gov/accessibility/checklist/).
- [x] Run `npm test` and make sure the tests for the files you have changed have passed.
- [ ] Run your code through [HTML_CodeSniffer](http://squizlabs.github.io/HTML_CodeSniffer/) and make sure it’s error free.
- [x] Title your pull request using this format: [Website] - [UI component]: Brief statement describing what this pull request solves.

### CODE REVIEW

**Review by mejiaj — CHANGES_REQUESTED** [review]
Good work, added some suggestions to tidy it up a bit.

File: src/stylesheets/components/_footer.scss [hunk 1]
```diff
@@ -1,5 +1,32 @@
-// General footer styles
+// Map for high contrast mode icons
+
+$icon-object: (
```
[the remark this answers is not in the document]
Comment (mahoneycm) [reply src/stylesheets/components/_footer.scss]: Resolved in [135349c](https://github.com/uswds/uswds/pull/4528/commits/135349c2930c6ca38f010ac2b2c30b705a1df891)

**Review by thisisdano — COMMENTED** [review]
LGTM once you address James's comments

File: src/stylesheets/components/_footer.scss [hunk 2]
```diff
@@ -298,6 +333,21 @@
...
       justify-content: flex-start;
 
+      @media (forced-colors: active) {
+        @include place-icon(
+          $chevron-expand-more,
+          "before",
+          $icon-margin,
+          $vertical-align,
+          $contrast-bg
+        );
```
[the remark this answers is not in the document]
Comment (mahoneycm) [reply src/stylesheets/components/_footer.scss]: Changed in [54e7665](https://github.com/uswds/uswds/pull/4528/commits/54e7665e05f8b58bff7046349e2d6d5c3f0b52c4)

**Comment by amyleadem** [comment]
@mahoneycm @mejiaj @thisisdano We will have to be careful of how these changes will play with the changes in PR #4476. I know we are working on merging both at around the same time, and it looks like there are different approaches to handling footer.scss. Flagging for awareness!

**Comment by amyleadem** [comment]
@mahoneycm I am concerned about the social media icon color contrast shown in your screenshots - I don't have great access to windows HC mode but I am wondering if it is possible to override either the background or icon color with `@media (forced-colors: active) and (prefers-color-scheme: light)`. This feels a bit like I'm trying to drive from the backseat though, so please disregard if not actually possible!

**Comment by mahoneycm** [comment]
Used `forced-color-adjust: none` to set the background color for the social link icons to lightgrey. This is easy to see in all high contrast modes and keeps things consistent between versions

image

File: src/stylesheets/components/_footer.scss [hunk 3]
```diff
@@ -309,9 +343,16 @@
         justify-content: center;
         margin-right: units(0.5);
         margin-left: units(-0.5);
+
+        @media (forced-colors: active) {
+          background-color: buttonText;
+          background-image: initial !important;
```
Comment (mejiaj) [inline src/stylesheets/components/_footer.scss:349]: `background-image: initial !important;` doesn't seem to do anything. Let's remove if not necessary.

File: src/stylesheets/components/_footer.scss [hunk 4]
```diff
@@ -324,13 +365,17 @@
...
       .usa-footer__primary-link {
-        &::before {
-          @include add-background-svg("usa-icons/navigate_next");
-        }
-
         @include at-media("mobile-lg") {
           margin: 0;
         }
+
+        @include place-icon($chevron-next, "before", 0.5, "auto", "default");
```
Comment (mejiaj) [inline src/stylesheets/components/_footer.scss:372]: Inlcude's should be first. Please move to `line 368` and have the media query come after.

File: src/stylesheets/components/_footer.scss [hunk 5]
```diff
@@ -309,9 +343,16 @@
         justify-content: center;
         margin-right: units(0.5);
         margin-left: units(-0.5);
+
+        @media (forced-colors: active) {
+          background-color: buttonText;
```
Comment (mejiaj) [inline src/stylesheets/components/_footer.scss:348]: You can set `!important` and remove the styles on `line 375`.

File: src/stylesheets/components/_footer.scss (hunk 3 above)
Comment (mahoneycm) [reply src/stylesheets/components/_footer.scss]: Resolved in [70eb57e](https://github.com/uswds/uswds/pull/4528/commits/70eb57e99dd316fb0fe0058fcb72e5d5456cd66a)

File: src/stylesheets/components/_footer.scss (hunk 4 above)
Comment (mahoneycm) [reply src/stylesheets/components/_footer.scss]: Resolved in [c1dae62](https://github.com/uswds/uswds/pull/4528/commits/c1dae62dac19c1f422211a5695d21d81ad914c57)

File: src/stylesheets/components/_footer.scss (hunk 5 above)
Comment (mahoneycm) [reply src/stylesheets/components/_footer.scss]: Resolved in [c1dae62](https://github.com/uswds/uswds/pull/4528/commits/c1dae62dac19c1f422211a5695d21d81ad914c57)

**Review by mejiaj — APPROVED** [review]
Looks good, thank you!

**Pull request merged**